spec/expression.dd

```diff
@@ -550,13 +550,15 @@ $(GNAME ShiftExpression):
         $(D >)$(D >)$(D >) is an unsigned right shift.
     )
 
-    $(P It's illegal to shift by the same or more bits than the size of the
-        quantity being shifted:)
+    $(UNDEFINED_BEHAVIOR
+    It is undefined behavior to shift by a negative value or by the same or more bits
+    than the size of the quantity being shifted:
```

Comment:
If bad shifts trigger undefined behavior, DMD needs to reject them in `@safe` code. It already does so when the right operand is a constant, but it needs to reject variables as well.

Reply:
I don't think undefined behaviour is the right term here, unspecified is. Undefined is for things that you have no ability to reason what state the program is in. `1 << x` where `x` is invalid gives you an `int`, you don't know what value it holds, but that in itself doesn't make the program state invalid.